src/tools/perf/perftest.c

```diff
@@ -733,7 +767,8 @@ static ucs_status_t setup_mpi_rte(struct perftest_context *ctx)
     rte_group_t group;
 
     rte_init(NULL, NULL, &group);
-    if (1 == rte_group_rank(group)) {
+    /* Let to the last rank to print the results */
+    if (rte_group_rank(group) == (rte_group_size(group) - 1)) {
```

Comment:
1. /* The last rank prints the results */
2. can move if after `#endif` (line 778) since it's same for HAVE_MPI and HAVE_RTE

Reply:
No, the "if" statements not the same if (rank == (size - 1)) { vs. if (rte_group_rank(group) == (rte_group_size(group) - 1)) {
in additional this is the original code, not mine

Reply:
they seem equivalent, maybe (rte_group_rank(group) == (rte_group_size(group) - 1)) can work for both cases

Reply:
Regarding line 738 - looks like there is a special reason the origin code checks it there in MPI case ONLY, in any case my patch isn't supposed to change/remove code if it is not required for loopback support...

Reply:
ok

Reply:
rte_group_rank ? maybe you mean rte_call(perf, group_index) ? the origin code does this check during rte group init, I don't see a reason to changer it